src/controllers/csi/provisioner/agent.go

```diff
@@ -98,17 +96,10 @@ func getUrlProperties(targetVersion, previousVersion string, pathResolver metada
 	}
 }
 
-func newImageInstaller(ctx context.Context, fs afero.Fs, apiReader client.Reader, pathResolver metadata.PathResolver, db metadata.Access, certPath string, dynakube *dynatracev1beta1.DynaKube) (installer.Installer, error) { //nolint:revive // argument-limit doesn't apply to constructors
+func newImageInstaller(ctx context.Context, fs afero.Fs, apiReader client.Reader, pathResolver metadata.PathResolver, db metadata.Access, dynakube *dynatracev1beta1.DynaKube) (installer.Installer, error) { //nolint:revive // argument-limit doesn't apply to constructors
 	dockerConfig := dockerconfig.NewDockerConfig(apiReader, *dynakube)
 	if dynakube.Spec.CustomPullSecret != "" {
-		err := dockerConfig.SetupAuths(ctx)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if dynakube.Spec.TrustedCAs != "" {
```

Comment:
You are now storing the trustedCAs only if there is a customPullSecret, that can lead to problems if a customer has trustedCAs but no customPullSecret

Reply:
You are right, this if should not exist here.

Added a unit test to check for this in the future.